crates/jsonschema/src/evaluation.rs

```diff
@@ -0,0 +1,1526 @@
+    if let Some(annotations) = &node.annotations {
+        state.serialize_field("annotations", annotations)?;
+    }
+    if let Some(annotations) = &node.dropped_annotations {
+        state.serialize_field("droppedAnnotations", annotations)?;
+    }
+    if !node.errors.is_empty() {
+        state.serialize_field("errors", &ErrorEntries(&node.errors))?;
+    }
+    if include_children && !node.children.is_empty() {
+        state.serialize_field(
+            "details",
+            &DetailsSerializer {
+                parent_path: evaluation_path,
+                children: &node.children,
+            },
+        )?;
+    }
+    state.end()
+}
+
+fn path_to_string(path: &[usize]) -> String {
+    if path.is_empty() {
+        String::new()
+    } else {
+        use itoa::Buffer;
+        // More accurate capacity estimate: '/' + average 2 digits per index
+        let mut result = String::with_capacity(path.len() * 3);
+        let mut buf = Buffer::new();
+        for segment in path {
+            result.push('/');
+            result.push_str(buf.format(*segment));
+        }
+        result
+    }
+}
+
+pub(crate) fn format_schema_location(
+    location: &Location,
+    absolute: Option<&Arc<Uri<String>>>,
+) -> String {
+    if let Some(uri) = absolute {
+        let base = uri.as_str();
+        if base.contains('#') {
+            base.to_string()
+        } else {
+            let loc_str = location.as_str();
+            if loc_str.is_empty() {
+                let mut result = String::with_capacity(base.len() + 1);
+                result.push_str(base);
+                result.push('#');
+                result
+            } else {
+                let mut result = String::with_capacity(base.len() + 1 + loc_str.len());
+                result.push_str(base);
+                result.push('#');
+                result.push_str(loc_str);
+                result
+            }
+        }
+    } else {
+        location.as_str().to_string()
+    }
+}
+
+struct ListEntry<'a> {
+    node: &'a EvaluationNode,
+    evaluation_path: String,
+}
+
+impl<'a> ListEntry<'a> {
+    fn new(node: &'a EvaluationNode, path: &[usize]) -> Self {
+        ListEntry {
+            node,
+            evaluation_path: path_to_string(path),
+        }
+    }
+}
+
+impl Serialize for ListEntry<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serialize_unit(self.node, &self.evaluation_path, serializer, false)
+    }
+}
+
+struct DetailsSerializer<'a> {
+    parent_path: &'a str,
+    children: &'a [EvaluationNode],
+}
+
+impl Serialize for DetailsSerializer<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(self.children.len()))?;
+        let mut buffer = String::new();
```

Comment:
Probably can be reused for more "details" fields. Maybe just general scratch buffer